In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv('Titanic_R.csv')
df.sample(10)

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 15 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   pclass     1309 non-null   int64  
 1   survived   1309 non-null   int64  
 2   Residence  1309 non-null   int64  
 3   name       1309 non-null   object 
 4   age        1046 non-null   float64
 5   sibsp      1309 non-null   int64  
 6   parch      1309 non-null   int64  
 7   ticket     1309 non-null   object 
 8   fare       1308 non-null   float64
 9   cabin      1309 non-null   object 
 10  embarked   1309 non-null   object 
 11  boat       1309 non-null   object 
 12  body       1309 non-null   object 
 13  home.dest  1309 non-null   object 
 14  Gender     1309 non-null   int64  
dtypes: float64(2), int64(6), object(7)
memory usage: 153.5+ KB


In [ ]:
# Changement dans la colonne age
df['age'].replace(' ', None, inplace=True)
df['age'] = df['age'].astype('float64')
df.info()

In [ ]:
# Changement dans la colonne fare
df['fare'].replace(' ', None, inplace=True)
df['fare'] = df['fare'].astype('float64')
df.info()

---
Q1. What is the average price paid for a ticket in the first class? 87.51

In [27]:
average_first_class = df[df['pclass']==1]['fare'].mean()
average_first_class

87.50899164086688

---
Q2. What is the average price paid for a ticket in the second class? 21.17

In [28]:
average_second_class = df[df['pclass']==2]['fare'].mean()
average_second_class

21.179196389891697

---
Q3. What is the average price paid for a ticket in the third class? 13.3

In [29]:
average_third_class = df[df['pclass']==3]['fare'].mean()
average_third_class

13.302888700564973

---
Q4. What is the average price paid by women for a ticket in the first class? 109.4

In [32]:
average_first_class_women = df[(df['pclass']==1) & (df['Gender']==1)]['fare'].mean()
average_first_class_women

109.41238472222221

---
Q5. What is the average price paid by men for a ticket in the third class? 12.4

In [33]:
average_third_class_men = df[(df['pclass']==3) & (df['Gender']==0)]['fare'].mean()
average_third_class_men

12.41546199186992

---
Q6. Who paid more for first class tickets on average, men or women? (Compare 109 for women and 69.88 for man) Women

In [34]:
average_first_class_men = df[(df['pclass']==1) & (df['Gender']==0)]['fare'].mean()
average_first_class_men

69.88838491620112

In [38]:
df.sample(10)

pclass  survived  Residence  \
131        1         1          0   
523        1         1          0   
34         3         1          2   
570        2         0          1   
659        1         1          0   
1298       3         1          2   
1085       2         0          1   
74         3         1          2   
607        3         0          0   
433        2         0          1   

                                                  name   age  sibsp  parch  \
131                                   Blank, Mr. Henry  40.0      0      0   
523                     Hays, Miss. Margaret Bechstein  24.0      0      0   
34                     Andersson, Miss. Erna Alexandra  17.0      4      2   
570                               Howard, Mr. Benjamin  63.0      1      0   
659   Kimball, Mrs. Edwin Nelson Jr (Gertrude Parsons)  45.0      1      0   
1298           Yasbeck, Mrs. Antoni (Selini Alexander)  15.0      1      0   
1085        Sedgwick, Mr. Charles Frederick Waddington  25.0      0      0   
74                                Ayoub, Miss. Banoura  13.0      0      0   
607                                Johansson, Mr. Nils  29.0      0      0   
433                                 Gale, Mr. Shadrach  34.0      1      0   

       ticket     fare cabin embarked boat body  \
131    112277  31.0000   A31        C    7        
523     11767  83.1583   C54        C    7        
34    3101281   7.9250              S    D        
570     24065  26.0000              S             
659     11753  52.5542   D19        S    5        
1298     2659  14.4542              C             
1085   244361  13.0000              S             
74       2687   7.2292              C    C        
607    347467   7.8542              S             
433     28664  21.0000              S             

                                 home.dest  Gender  
131                         Glen Ridge, NJ       0  
523                           New York, NY       1  
34    Ruotsinphyhtaa, Finland New York, NY       1  
570                       Swindon, England       0  
659                             Boston, MA       1  
1298                                             1  
1085                             Liverpool       0  
74                    Syria Youngstown, OH       1  
607                                              0  
433             Cornwall / Clear Creek, CO       0

---
Q7. Who paid more for second class tickets on average, American or British passengers? British

In [39]:
average_second_class_america = df[(df['pclass']==2) & (df['Residence']==0)]['fare'].mean()
average_second_class_america

17.157935714285713

In [41]:
average_second_class_british = df[(df['pclass']==2) & (df['Residence']==1)]['fare'].mean()
average_second_class_british

22.583823529411767

In [42]:
average_second_class_other = df[(df['pclass']==2) & (df['Residence']==2)]['fare'].mean()
average_second_class_other

20.95559696969697

---
Q8. How many people were on board with home residence neither American nor British? 749

In [46]:
nb_people_other = len(df[df['Residence']==2])
nb_people_other

749

---
Q9. On average, how old were men on board? What about men on board with third class tickets? 30.59, 25.96

In [48]:
average_men = df[df['Gender']==0]['age'].mean()
average_men

30.5852329787234

In [49]:
average_men_third_class = df[(df['Gender']==0) & (df['pclass']==3)]['age'].mean()
average_men_third_class

25.96227306590258

---
Q10. Which group was the youngest, first-, second-, or third-class? 3rd class

In [50]:
average_first_class = df[df['pclass']==1]['age'].mean()
average_first_class

39.15991795774648

In [51]:
average_second_class = df[df['pclass']==2]['age'].mean()
average_second_class

29.506704980842912

In [52]:
average_third_class = df[df['pclass']==3]['age'].mean()
average_third_class

24.81636726546906

---
Q11. What is the average family size in the second class if you know that the data contains only the main passenger name? 0.76

In [55]:
df['family'] = df['sibsp'] + df['parch'] + 1
average_family_size = df['family'].mean()
average_family_size

1.8838808250572956

---
Q12. On average, which ticket group had the largest family on board? 3rd

In [56]:
average_family_first_class = df[df['pclass']==1]['family'].mean()
average_family_first_class

1.8018575851393188

In [57]:
average_family_second_class = df[df['pclass']==2]['family'].mean()
average_family_second_class

1.7617328519855595

In [58]:
average_family_third_class = df[df['pclass']==3]['family'].mean()
average_family_third_class

1.9689703808180536